src/encoded/static/components/item-pages/HiGlassViewConfigView.js

```diff
@@ -647,9 +670,28 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
         );
     }
 
+    onViewConfigUpdated(viewConf){
+        const { tilesetUids: oldData } = this.state;
+        const newData = this.memoized.getTilesetUids(JSON.parse(viewConf));
```

Comment:
This is memoized but the object passed in will always be brand new reference (output of `JSON.parse()`); if you want to compare deeply nested values could supply a [custom "equality" function](https://www.npmjs.com/package/memoize-one#custom-equality-function) as 2nd param in `memoize(funcToMemoize, customCompareFunction)`.

Could alternatively try to do the JSON.parse within `getTilesetUids(textViewConf)`, but would be worried that performance of comparing such long text strings would not be performant... idk though, would be interesting to profile.

If `onViewConfigUpdated` isn't called frequently, it should be ok to not have it memoized.
If is an event handler and it can theoretically be called frequently and there is concern it would "freeze" the synchronous JS execution, could use approaches like `setTimeout` or [`_.debounce`](https://underscorejs.org/#debounce) (`this.onViewConfigUpdated = _.debounce(this.onViewConfigUpdated.bind(this), 750);`).

Reply:
@alexkb0009  `setTimeout` or `_.debounce (this.onViewConfigUpdated = _.debounce(this.onViewConfigUpdated.bind(this), 750);)` is a good approach! Will try it and merge then. Thanks!